### ISSUE TITLE
PearsonCorrcoef breaks when a single sample is passed at a time

### ISSUE THREAD

**Issue by raimis** [PyTorchLightning/metrics#274]
## 🐛 Bug

When passing a single sample to the PearsonCorrcoef metric, it will crash. Similar to #227 and partially fixed by #229. Originally reported in https://github.com/PyTorchLightning/metrics/issues/227#issuecomment-843046691

### To Reproduce

```python
import torch
import torchmetrics

print(torch.__version__)
print(torchmetrics.__version__)

correlation = torchmetrics.PearsonCorrcoef()

correlation.update(torch.tensor([0.3]), torch.tensor([0.4]))
correlation.update(torch.tensor([0.7]), torch.tensor([0.5]))
correlation.update(torch.tensor([0.9]), torch.tensor([0.4]))
correlation.compute()
```

Output:
```
1.8.0
0.3.2
/shared/raimis/opt/miniconda/envs/tmp/lib/python3.8/site-packages/torchmetrics/utilities/prints.py:36: UserWarning: Metric `PearsonCorrcoef` will save all targets and predictions in buffer. For large datasets this may lead to large memory footprint.
  warnings.warn(*args, **kwargs)
Traceback (most recent call last):
  File "bug.py", line 12, in <module>
    correlation.compute()
  File "/shared/raimis/opt/miniconda/envs/tmp/lib/python3.8/site-packages/torchmetrics/metric.py", line 251, in wrapped_func
    self._computed = compute(*args, **kwargs)
  File "/shared/raimis/opt/miniconda/envs/tmp/lib/python3.8/site-packages/torchmetrics/regression/pearson.py", line 95, in compute
    preds = torch.cat(self.preds, dim=0)
RuntimeError: zero-dimensional tensor (at position 0) cannot be concatenated
```

### Expected behavior

The metric is computed without throwing the `RuntimeError`.

**Comment by edgarriba** [comment]
@raimis please try to update `torchmetrics>0.3.2` - in master this issue is fixed.

**Comment by edgarriba** [comment]
I guess at some point soon since last release was a month ago. Any insights here @Borda @edenlightning ?

**Comment by Borda** [comment]
> Do you know when the next version will be released?

do you need anything specific from master?

**Comment by raimis** [comment]
> do you need anything specific from master?

Just the fix of this issue.

**Comment by Borda** [comment]
I see we can make one more this week if you want :]

**Comment by raimis** [comment]
I tried again:

- Install `torchmetrics` with `conda`:
  ```bash
  $ conda create -n tmp_1 -c conda-forge torchmetrics pytorch-gpu
  ```
- The version of `torchmetrics` is `0.3.2`, as expected:
  ```bash
  $conda activate tmp_1
  $ conda list | grep torchmetrics
  torchmetrics              0.3.2              pyhd8ed1ab_0    conda-forge
  ```
- Run the code above and get the same error:
  ```python
  $ python
  Python 3.9.4 | packaged by conda-forge | (default, May 10 2021, 22:13:33) 
  [GCC 9.3.0] on linux
  Type "help", "copyright", "credits" or "license" for more information.
  >>> import torch
  >>> import torchmetrics
  >>> print(torch.__version__)
  1.8.0
  >>> print(torchmetrics.__version__)
  0.3.2
  >>> correlation = torchmetrics.PearsonCorrcoef()
  /shared/raimis/opt/miniconda/envs/tmp_1/lib/python3.9/site-packages/torchmetrics/utilities/prints.py:36: UserWarning: Metric `PearsonCorrcoef` will save all targets and predictions in buffer. For large datasets this may lead to large memory footprint.
  warnings.warn(*args, **kwargs)
  >>> correlation.update(torch.tensor([0.3]), torch.tensor([0.4]))
  >>> correlation.update(torch.tensor([0.7]), torch.tensor([0.5]))
  >>> correlation.update(torch.tensor([0.9]), torch.tensor([0.4]))
  >>> correlation.compute()
  Traceback (most recent call last):
    File "<stdin>", line 1, in <module>
    File "/shared/raimis/opt/miniconda/envs/tmp_1/lib/python3.9/site-packages/torchmetrics/metric.py", line 251, in wrapped_func
      self._computed = compute(*args, **kwargs)
    File "/shared/raimis/opt/miniconda/envs/tmp_1/lib/python3.9/site-packages/torchmetrics/regression/pearson.py", line 95, in compute
      preds = torch.cat(self.preds, dim=0)
  RuntimeError: zero-dimensional tensor (at position 0) cannot be concatenated
  >>> 
  ```
So, the issues is not fixed in `0.3.2`.

**Comment by SkafteNicki** [comment]
Fix included in v0.4.0 just released (https://github.com/PyTorchLightning/metrics/releases/tag/v0.4.0). Closing.

**Comment by raimis** [comment]
Thanks!
